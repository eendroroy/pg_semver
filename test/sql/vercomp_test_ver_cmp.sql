LOAD 'vercomp';

SELECT VERSION_CMP('0.0.0', '0.0.1');
SELECT VERSION_CMP('0.0.1', '0.0.1');
SELECT VERSION_CMP('0.0.2', '0.0.1');

SELECT VERSION_CMP('1.0.0-alpha1', '1.0.0-alpha0');
SELECT VERSION_CMP('1.0.0-alpha1', '1.0.0-alpha2');
SELECT VERSION_CMP('1.0.0-alpha1', '1.0.0-beta2');
SELECT VERSION_CMP('1.0.0-rc1', '1.0.0');

SELECT VERSION_CMP('1.0.0-rc1*', '1.0.0');
SELECT VERSION_CMP('1.0.0-rc@1', '1.0.0');
SELECT VERSION_CMP('1.0.0-rc#1', '1.0.0');
