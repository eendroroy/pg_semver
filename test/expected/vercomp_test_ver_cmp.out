LOAD 'vercomp';
SELECT VER_CMP('0.0.0'::VER, '0.0.1'::VER);
 ver_cmp 
---------
      -1
(1 row)

SELECT VER_CMP('0.0.1'::VER, '0.0.1'::VER);
 ver_cmp 
---------
       0
(1 row)

SELECT VER_CMP('0.0.2'::VER, '0.0.1'::VER);
 ver_cmp 
---------
       1
(1 row)

SELECT VER_CMP('1.0.0-alpha1'::VER, '1.0.0-alpha0'::VER);
 ver_cmp 
---------
       1
(1 row)

SELECT VER_CMP('1.0.0-alpha1'::VER, '1.0.0-alpha2'::VER);
 ver_cmp 
---------
      -1
(1 row)

SELECT VER_CMP('1.0.0-alpha1'::VER, '1.0.0-beta2'::VER);
 ver_cmp 
---------
      -1
(1 row)

SELECT VER_CMP('1.0.0-rc1'::VER, '1.0.0'::VER);
 ver_cmp 
---------
      -1
(1 row)

SELECT VER_CMP('1.0.0-rc1*'::VER, '1.0.0'::VER);
ERROR:  Invalid Version 1.0.0-rc1*.
LINE 1: SELECT VER_CMP('1.0.0-rc1*'::VER, '1.0.0'::VER);
                       ^
SELECT VER_CMP('1.0.0-rc@1'::VER, '1.0.0'::VER);
ERROR:  Invalid Version 1.0.0-rc@1.
LINE 1: SELECT VER_CMP('1.0.0-rc@1'::VER, '1.0.0'::VER);
                       ^
SELECT VER_CMP('1.0.0-rc#1'::VER, '1.0.0'::VER);
ERROR:  Invalid Version 1.0.0-rc#1.
LINE 1: SELECT VER_CMP('1.0.0-rc#1'::VER, '1.0.0'::VER);
                       ^
